packages/drivers/odsp-driver/src/contracts.ts

```diff
@@ -35,7 +35,10 @@ export interface IOdspResolvedUrl extends IFluidResolvedUrl {
 
     summarizer: boolean;
 
-    sharingLinkP?: Promise<string>;
+    sharingLinkOptions?: {
```

Comment:
I think we want comments here. I'm not super familiar with the exact nuance between the Teams share link vs generated share links, but an appGeneratedSharingLink doesn't feel right here

Reply:
@ram-ramasubramanian - do you have a good name for the kind of share link Teams is providing here? Obviously the share links that the driver generate don't apply to this flow, so I'm guessing not all kinds of "sharing links" passed in by the host app are eligible for this flow

Reply:
Perhaps we should just make this sharingLinkToRedeem or something like that and not make it a promise since that should only ever be passed in by the host? I'm not fully familiar with the "generated" share link flow and if it should be on the resolved URL interface. Thoughts @leeviana?

Reply:
Renamed to redeemSharingLink and changed from promise to just a string on resolved url. The cache logic is now same as it was before.